src/main/java/nextstep/ladder/domain/Direction.java

```diff
@@ -0,0 +1,82 @@
+package nextstep.ladder.domain;
+
+import nextstep.ladder.strategy.ConnectStrategy;
+import nextstep.ladder.strategy.RandomConnectStrategy;
+
+import java.util.Objects;
+
+public class Direction {
+    private static final String CHECK_CONTIONUOUS = "가로 라인이 겹치는 지 확인해주세요.";
+    private final boolean left;
+    private final boolean right;
+
+    public Direction(boolean left, boolean right) {
+        validateContinuous(left, right);
+        this.left = left;
+        this.right = right;
+    }
+
+    private void validateContinuous(boolean left, boolean right) {
+        if (left & right) {
+            throw new IllegalArgumentException(CHECK_CONTIONUOUS);
+        }
+    }
+
+    public static Direction first(ConnectStrategy connectStrategy) {
+        if (connectStrategy.isConnectable()) {
+            return new Direction(false, true);
+        }
+
+        return new Direction(false, false);
+    }
+
+    public Direction next() {
+        if (right) {
+            return next(() -> false);
+        }
+
+        return next(new RandomConnectStrategy());
+    }
+
+    public Direction next(ConnectStrategy connectStrategy) {
+        if (connectStrategy.isConnectable()) {
+            return new Direction(right, true);
+        }
+
+        return new Direction(right, false);
+    }
+
+    public Direction last() {
+        return new Direction(right, false);
+    }
+
+    public boolean hasAnyDirection() {
+        return left | right;
+    }
```

Comment:
```suggestion
    public boolean hasAnyDirection() {
        return left || right;
    }
```
어차피 left, right가 boolean 이기 때문에 이렇게 표현해야 하지 않을까요?

Reply:
아.. 비트연산자를 사용했네요... 
해당 부분 수정하였습니다 감사합니다!